frontend/pages/browse.js

```diff
@@ -123,46 +127,71 @@ export default function Index({
     if (params.message) setMessage(decodeURI(params.message));
     if (params.errorMessage) setErrorMessage(decodeURI(params.errorMessage));
   });
-  const [tabValue, setTabValue] = React.useState(hash ? typesByTabValue.indexOf(hash) : 0);
-  const [filtersExpanded, setFiltersExpanded] = React.useState(false);
-  const [filters, setFilters] = React.useState({
+
+  const [tabValue, setTabValue] = useState(hash ? typesByTabValue.indexOf(hash) : 0);
+  const [filtersExpanded, setFiltersExpanded] = useState(false);
+
+  const [filters, setFilters] = useState({
     projects: {},
     members: {},
     organizations: {},
   });
 
+  /**
+   * When applying new filters, update the state of the
+   * loading spinner until the async call finishes.
+   */
   const applyNewFilters = async (type, newFilters, closeFilters) => {
-    if (filters !== newFilters) {
-      setFilters({ ...filters, [type]: newFilters });
-      const newUrlEnding = buildUrlEndingFromFilters(newFilters);
-      if (state.urlEnding[type] != newUrlEnding) {
-        if (closeFilters) setFiltersExpanded(false);
-        try {
-          let filteredItemsObject;
-          if (type === "projects") filteredItemsObject = await getProjects(1, token, newUrlEnding);
-          else if (type === "organizations")
-            filteredItemsObject = await getOrganizations(1, token, newUrlEnding);
-          else if (type === "members") {
-            filteredItemsObject = await getMembers(1, token, newUrlEnding);
-            filteredItemsObject.members = membersWithAdditionalInfo(filteredItemsObject.members);
-          }
-          setState({
-            ...state,
-            items: { ...state.items, [type]: filteredItemsObject[type] },
-            hasMore: { ...state.hasMore, [type]: filteredItemsObject.hasMore },
-            urlEnding: { ...state.urlEnding, [type]: newUrlEnding },
-            nextPages: { ...state.nextPages, [type]: 2 },
-          });
-        } catch (e) {
-          console.log(e);
-        }
+    // Short circuit if filters are equal
+    if (filters == newFilters) {
+      return;
+    }
+
+    setFilters({ ...filters, [type]: newFilters });
+
+    const newUrlEnding = buildUrlEndingFromFilters(newFilters);
+    if (state.urlEnding[type] == newUrlEnding) {
```

Comment:
Additional suggestion:
```suggestion
    if (state.urlEnding[type] === newUrlEnding) {
```